### ISSUE TITLE
Manage all App Configuration data in one file

### ISSUE THREAD

**Issue by realbart** [Azure/AppConfiguration#552]
I would very much like to be able to create/update app configuration values, with different labels and also keyvault references in one step. At this moment I'm forced to make a separate file AND pipeline step for every label/contenttype-combination.

I would expect the config file to look a bit like this:

```
[
  {
    "label": "My.Label"
    "key": "MySetting:Foo",
    "value": "FooValue",
  },
  {
    "key": "ConnectionStrings:Bar",
    "value": "https://tdm-bar.azure.net/secrets/Bar/",
    "contenttype": "application/vnd.microsoft.appconfig.keyvault"
  }
]
```

Our company policy says all application code and configuration must originate from either source control or the release pipeline (which we are also putting in source control). 
In other words: no setting is to be done manually and no setting can be done without review by someone else.

**Comment by zhenlan** [comment]
@realbart, great feedback. The design of the current file import/export is to enable users to consume those config files in their applications without any code change. What you bring up is a bit different scenario where you need to use files for configuration authoring. It's absolutely a reasonable scenario to me. We will look into it.

**Comment by zhenlan** [comment]
We add the capability in **Azure CLI** so you can manage your App Configuration key-values with different labels, tags, content-types all in one JSON file. Please give it a try and share any feedback with us. If it works well, we will add the capability in other tools, such as Azure pipeline tasks, GitHub actions, portal, etc.

You will need to download [Azure CLI](https://docs.microsoft.com/cli/azure/install-azure-cli) version **2.30.0** or later. Then use parameter **--profile appconfig/kvset** when importing from or exporting to a file. For example,
```cli
az appconfig kv export -n <store_name> -d file --path <file_path> --format json --profile appconfig/kvset
az appconfig kv import -n <store_name> -s file --path <file_path> --format json --profile appconfig/kvset
```

**Comment by realbart** [comment]
Nice!

I just tried it on CloudShell and it does exactly what I need.
I’ll try and integrate it into our pipeline tomorrow, initially just with an Azure CLI Task<https://docs.microsoft.com/en-us/azure/devops/pipelines/tasks/deploy/azure-cli?view=azure-devops>.

This makes things a lot easier for us.
Thanks!

Bart Kemps

**Comment by HugoSousa** [comment]
> We add the capability in **Azure CLI** so you can manage your App Configuration key-values with different labels, tags, content-types all in one JSON file. Please give it a try and share any feedback with us. If it works well, we will add the capability in other tools, such as Azure pipeline tasks, GitHub actions, portal, etc.
> 
> You will need to download [Azure CLI](https://docs.microsoft.com/cli/azure/install-azure-cli) version **2.30.0** or later. Then use parameter **--profile appconfig/kvset** when importing from or exporting to a file. For example,
> 
> ```
> az appconfig kv export -n <store_name> -d file --path <file_path> --format json --profile appconfig/kvset
> az appconfig kv import -n <store_name> -s file --path <file_path> --format json --profile appconfig/kvset
> ```

It seems that this does not allow to export to a file for all labels in a single shot. I am only able to export a single label at a time. Reading the documentation for flag _--label_, it says:

_Only keys and feature flags with this label will be exported. If no label specified, export keys and feature flags with null label by default. **Only when export destination is appconfig**, we support star sign as filters, for instance * means all labels and abc* means labels with abc as prefix. Label filters are not supported when exporting to file or appservice._

Could this be the case when exporting to a file also?

**Comment by zhenlan** [comment]
@HugoSousa give it a try with `--label *`. It should export all labels at once. We realized we missed updating the help message you mentioned above. It will be updated in the next CLI release. I also updated my message above inline.

**Comment by zhenlan** [comment]
We have an official document on this topic out now
https://docs.microsoft.com/azure/azure-app-configuration/concept-config-file

**Comment by MaryanneNjeri** [comment]
A new release is out [Azure App configuration Push task version 3.3.0](https://marketplace.visualstudio.com/items?itemName=AzureAppConfiguration.azure-app-configuration-task-push) allows you to manage your App Configuration key-values with different labels, tags, content-types all in one JSON file. Please give it a try and share feedback with us. 
You will need to specify the FileContentProfile as `appconfig/kvset`. Check out the[ schema](https://github.com/Azure/AppConfiguration/blob/main/docs/KVSet/KVSet.v1.0.0.schema.json)

**Comment by jimmyca15** [comment]
This enhancement has been added.
